src/testparameters.js

```diff
@@ -21,54 +21,227 @@
+      });
+      const response = await apiEndpointPrompt(allApiEndpointsString);
+      const apiEndpointsString = response.apiEndpoints;
+      apiEndpoints = apiEndpointsString.map(function(apiEndpoint) {
+        return JSON.parse(apiEndpoint);
+      });
+      extraConfig.apiEndpoints = apiEndpoints;
+    }
+  }
+
+  const apiKeysObject = {};
+  apiKeys.forEach(function(apiKey) {
+    apiKeysObject[apiKey.name] = apiKey.value;
+  });
+  config.apiKeys = config.apiKeys || [];
+  config.apiKeys.forEach(function(apiKey) {
+    /*
+      Fetch the value of API key from config file, only if
+      it is not provided by the user through command options.
+    */
+    if (!apiKeysObject[apiKey.name]) apiKeysObject[apiKey.name] = apiKey.value;
+  });
+  const requiredApiKeys = getApiKeyList(apiEndpoints, oasDoc);
+  for (const requiredApiKeyName of requiredApiKeys) {
+    const response = await apiKeyPrompt(requiredApiKeyName);
+    apiKeysObject[requiredApiKeyName] = response.apiKeyValue;
+    extraConfig.apiKeys = extraConfig.apiKeys || [];
+    extraConfig.apiKeys.push({
+      name: requiredApiKeyName,
+      value: response.apiKeyValue,
+    });
+  }
+
+  if (isBasicAuthRequired(apiEndpoints, oasDoc) && basicAuth == null) {
+    if (config.basicAuth) basicAuth = config.basicAuth;
+    else {
+      let response;
+      response = await usernamePrompt();
+      basicAuth = {};
+      basicAuth.userName = response.userName;
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        response = await passwordPrompt();
+        if (response.password !== response.confirmPassword) {
+          logger.error('password and confirm password does not match'.red);
+        } else {
+          basicAuth.password = response.password;
+          break;
+        }
+      }
+      extraConfig.basicAuth = basicAuth;
+    }
+  }
+
+  if (timeout == null && config.timeout) {
+    timeout = config.timeout;
+  }
 
-  let basicAuthCredentials;
-  if (isBasicAuthRequired(apiEndpoints, testSuiteFile.oasDoc)) {
-    // Get basic auth credentials from the user.
-    // Below basic auth credentials is a dummy value used for demo purpose.
-    basicAuthCredentials = {
-      'username': 'sunar',
-      'password': 'asd23ad%23',
-    };
+  if (!_.isEmpty(extraConfig)) {
+    const response = await upsertConfigPrompt();
+    if (response.upsertConfig) {
+      const keys = Object.keys(extraConfig);
+      keys.forEach(function(key) {
+        if (key == 'apiKeys') {
+          /*
+            In order to persist the exisiting API keys present in config file ,
+            we concat the new API keys to it.
+          */
+          config[key] = config[key] || [];
+          config[key] = config[key].concat(extraConfig[key]);
+          return;
+        }
+        config[key] = extraConfig[key];
+      });
+      const response = await configDetailsPrompt();
+      const configDirectoryPath = response.configDirectoryPath;
```

Comment:
Redundant, can use `response.configDirectoryPath` directly

Reply:
Removed redundant piece of code.